EngineLayer/MetaMorpheusEngine.cs

```diff
@@ -41,98 +41,161 @@ protected MetaMorpheusEngine(List<string> nestedIds)
 
         #region Public Methods
 
-        public static double MatchIons(IMsDataScan<IMzSpectrum<IMzPeak>> thisScan, Tolerance productMassTolerance, double[] sorted_theoretical_product_masses_for_this_peptide, double[] matchedIonMassesListPositiveIsMatch, bool addComp, double precursorMass, List<ProductType> lp)
+        public static void MatchIons(IMsDataScan<IMzSpectrum<IMzPeak>> thisScan, Tolerance productMassTolerance, double[] sortedTheoreticalProductMassesForThisPeptide, List<double> matchedIonMassesList, List<double> productMassErrorDa, List<double> productMassErrorPpm, bool addComp, double precursorMass, List<ProductType> lp)
         {
-            var TotalProductsHere = sorted_theoretical_product_masses_for_this_peptide.Length;
+            var TotalProductsHere = sortedTheoreticalProductMassesForThisPeptide.Length;
             if (TotalProductsHere == 0)
-                return 0;
+                return;
             int MatchingProductsHere = 0;
             double MatchingIntensityHere = 0;
 
             // speed optimizations
             double[] experimental_mzs = thisScan.MassSpectrum.XArray;
             double[] experimental_intensities = thisScan.MassSpectrum.YArray;
 
-            if (addComp)
+            if(addComp)
                 AddComplementaryPeaks(ref experimental_mzs, ref experimental_intensities, precursorMass, lp);
 
-            int num_experimental_peaks = experimental_mzs.Length;
+            int numExperimentalPeaks = experimental_mzs.Length;
 
             int currentTheoreticalIndex = -1;
             double currentTheoreticalMass;
             do
             {
                 currentTheoreticalIndex++;
-                currentTheoreticalMass = sorted_theoretical_product_masses_for_this_peptide[currentTheoreticalIndex];
-            } while (double.IsNaN(currentTheoreticalMass) && currentTheoreticalIndex < sorted_theoretical_product_masses_for_this_peptide.Length - 1);
+                currentTheoreticalMass = sortedTheoreticalProductMassesForThisPeptide[currentTheoreticalIndex];
+            } while (double.IsNaN(currentTheoreticalMass) && currentTheoreticalIndex < sortedTheoreticalProductMassesForThisPeptide.Length - 1);
 
             if (double.IsNaN(currentTheoreticalMass))
-                return 0;
+                return;
 
             double currentTheoreticalMz = currentTheoreticalMass + Constants.protonMass;
 
-            int testTheoreticalIndex;
```

Comment:
flipped order here. Used to skip theoreticals until one was close to an experimental, but experimentals vastly outnumber theoreticals, so this speeds up the loop.

Reply:
Please run a 3-task (classic search+gptmd+classic search) procedure on 3 different spectra files, including both qhf and velos files. Both before and after this change, but run on the same computer. Maybe Artemis? Then please give me the AllResults.txt for all six runs. 

This is a fundamental change to the search procedure, and I want to make sure it didn't make things slower.

Reply:
Will do.